source/initial-dns-seedlist-discovery/tests/README.rst

```diff
@@ -118,27 +118,35 @@ These YAML and JSON files contain the following fields:
 .. _`Connection String`: ../../connection-string/connection-string-spec.rst
 .. _`URI options`: ../../uri-options/uri-options.rst
 
-For each file, create a MongoClient initialized with the ``mongodb+srv``
-connection string.
-
-If ``seeds`` is specified, drivers SHOULD verify that the set of hosts in the
-client's initial seedlist matches the list in ``seeds``. If ``numSeeds`` is
-specified, drivers SHOULD verify that the size of that set matches ``numSeeds``.
-
-If ``hosts`` is specified, drivers MUST verify that the set of
-ServerDescriptions in the client's TopologyDescription eventually matches the
-list in ``hosts``. If ``numHosts`` is specified, drivers MUST verify that the
-size of that set matches ``numHosts``.
-
-If ``options`` is specified, drivers MUST verify each of the values under
-``options`` match the MongoClient's parsed value for that option. There may be
-other options parsed by the MongoClient as well, which a test does not verify.
-
-If ``parsed_options`` is specified, drivers MUST verify that each of the values
-under ``parsed_options`` match the MongoClient's parsed value for that option.
-Supported values include, but are not limited to, ``user`` and ``password``
-(parsed from ``UserInfo``) and ``auth_database`` (parsed from
-``Auth database``).
-
-If ``error`` is specified and ``true``, drivers MUST verify that an error has
-been thrown.
+For each YAML file:
+
+- Create a MongoClient initialized with the ``mongodb+srv``
+  connection string.
+- Run run a ping operation.
```

Comment:
@kmahar sorry for the really slow response! I looked into this and it turns out the Go driver doesn't perform any kind of auth in the initial DNS seedlist discovery test. The Go driver doesn't run a `ping`, it only creates an unauthenticated connection to the server (like the connections used for the RTT monitor).

So far I'm assuming that most drivers don't currently perform any operations that require auth and are not actually testing initial DNS seedlist discovery with authentication, so performing some kind of authentication in the tests would essentially be covering a new use case. Adding coverage for testing initial DNS seedlist discovery with auth enabled isn't the intent of this change and create the user from the `MONGODB_URI` connection string sounds like a decent amount of work, so I think we should further amend the spec to suggest or require running only on deployments without auth. If we determine that running the tests on deployments with auth (where the driver actually performs auth handshake) is desired additional coverage, we should make that change separately.

I'll make the proposed edits to the test description.

Reply:
Hmm, I agree that logically there's not really a reason to run the tests with auth. That said I think either approach is going to incur some undesirable additional work... I believe most drivers test with either both or neither of auth and TLS enabled (at least I know this is true for Swift, Rust, Node, Java.) Which I think is mainly just to cut down on the total number of tasks we need to run.

Since almost all of the DNS seedlist tests require TLS, most drivers probably only currently run them with auth also on. So if we want to tell drivers to run with TLS and no auth, there would also be some work involved for any driver with that setup to start testing against TLS-only configs so they still get coverage of these tests. 

I'm not really sure which I prefer... I think your proposal is the more "correct" approach, but I'm hesitant to ask everyone to blow up their Evergreen configs with even more tasks just for the sake of this test runner. Would be curious to get some more opinions

Reply:
I would like not to blow up Java driver testing with yet more variants.

Reply:
I've updated the proposed changes here to add a new `ping` boolean to the test format which specifies whether to attempt running a "ping" operation. The "ping" operation doesn't require authentication, so it _shouldn't_ require drivers to do any of the authentication work described by @kmahar or change the Evergreen config any.

I've validated that these changes work in the Go driver. I'm going to ask someone to test the changes in at least one more driver before requesting re-review of this PR.

Reply:
@matthewdale I tried these out in Rust. I'm running into an issue we may have discussed before (it's been a while 😅) - some of the test files, such as `encoded-userinfo-and-db`, and `uri-with-auth`, include credentials in the connection string. so while auth is not required to run a ping, the MongoClient we create using the connection string attempts to authenticate with the specified username and password when handshaking connections, and fails to do so, which prevents us from running the ping via `Database.run_command`. how do you run the ping in Go to bypass the auth info? 

there are workarounds (like the one we currently have that I describe above, to pre-create the connection string's user), or to manipulate the parsed connection string to remove the username and password before creating the client, but neither of these seem great.  maybe the tests including auth info could omit `ping: true`?

Reply:
Ah, great find! I also forgot what tests were problematic in other drivers, so that's great you were able to identify just the ones with provided but incorrect userinfo. I'll set `ping: false` in those tests and provide a comment describing why we don't run a "ping".

I actually don't know how running a "ping" using a URI with provided but incorrect userinfo succeeds in the Go driver. I'll look into it more.

Reply:
tested in Rust and they pass with these changes 👍

Reply:
Also, I don't know why I thought the "with auth" tests pass in Go, because they fail when running a "ping". The Go Driver `master` branch currently doesn't include those two tests.